uni/lib/view/login/login.dart

```diff
@@ -230,8 +242,50 @@ class LoginPageViewState extends State<LoginPageView> {
     if (status == RequestStatus.successful && session.authenticated) {
       Navigator.pushReplacementNamed(
           context, '/${DrawerItem.navPersonalArea.title}');
-    } else if (status == RequestStatus.failed) {
-      ToastMessage.error(context, 'O login falhou');
     }
   }
+
+  void updatePasswordDialog() {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text("A tua palavra-passe expirou"),
+          content: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              Text(
+                  'Por razões de segurança, as palavras-passes têm de ser alteradas periodicamente.',
+                  textAlign: TextAlign.start,
+                  style: Theme.of(context).textTheme.titleSmall),
+              const SizedBox(height: 20),
+              const Align(
+                  alignment: Alignment.centerLeft,
+                  child: Text(
+                    'Deseja alterar a palavra-passe?',
+                    textAlign: TextAlign.start,
+                  )),
+            ],
+          ),
+          actions: [
+            TextButton(
+              child: const Text("Cancelar"),
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+            ),
+            ElevatedButton(
+              child: const Text("Alterar"),
+              onPressed: () async {
+                const url = "https://self-id.up.pt/password";
+                if (await canLaunchUrl(Uri.parse(url))) {
+                  await launchUrl(Uri.parse(url));
+                }
```

Comment:
What if it can't and the user has just clicked the button? Should we flash a toast?
(Also, this seems to be recurrent across the app. Should we encapsulate that behaviour?)

Reply:
I didn't do it because I believe there is an issue to do that